### PR TITLE
add_tests to regions controler cover 100%

### PR DESCRIPTION
**Pull request by sharix4**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo tests to regions controler and added several tests to other controllers

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: EPlast/EPlast.Tests/Controllers/RegionsControllerTests.cs [hunk 1]
```diff
@@ -613,6 +632,203 @@ public async Task RemoveDocument_Id_ReturnsOkResult()
...
+        {
+
+            // Arrange
+            _userManager.Setup(x => x.GetUserAsync(It.IsAny<ClaimsPrincipal>())).ReturnsAsync(new User());
+            _userManager.Setup(x => x.GetRolesAsync(It.IsAny<User>())).ReturnsAsync(new List<string>() { "Admin" });
+
+            _regionAnnualReportService.Setup(x => x.GetRegionMembersInfo(It.IsAny<int>(), It.IsAny<int>()))
+            .ReturnsAsync(new List<RegionMembersInfo>());
+            int reportID = 1;
+            // Act
```
Comment (danylo-melnychuk) [inline EPlast/EPlast.Tests/Controllers/RegionsControllerTests.cs:681]: //act in the new line

File: EPlast/EPlast.Tests/Controllers/RegionsControllerTests.cs [hunk 2]
```diff
@@ -613,6 +632,203 @@ public async Task RemoveDocument_Id_ReturnsOkResult()
...
+            mockHttpContext.Setup(m => m.User).Returns(new ClaimsPrincipal());
+
+            _regionController.ControllerContext.HttpContext = mockHttpContext.Object;
+
+            byte[] bytes = new byte[2];
+            _cache.Setup(x => x.GetAsync(It.IsAny<string>(), default)).ReturnsAsync((byte[])null);
+       
+            // Act
+            var result = await _regionController.GetRegions(page, pageSize, regionName);
+            // Assert
```
Comment (danylo-melnychuk) [inline EPlast/EPlast.Tests/Controllers/RegionsControllerTests.cs:827]: pls put enter between Act and Assert

File: EPlast/EPlast.Tests/Controllers/RegionsControllerTests.cs [hunk 3]
```diff
@@ -481,6 +482,24 @@ public async Task GetRegionsBoardAsync_ReturnsRegionsBoard()
...
+            // Arrange
+            _userManager
+                .Setup(x => x.GetUserAsync(new System.Security.Claims.ClaimsPrincipal())).ReturnsAsync(new User());
+            _regionService
+                .Setup(x => x.GetRegionByNameAsync(EnumExtensions.GetDescription(RegionsStatusType.RegionBoard), It.IsAny<User>()))
+                .ReturnsAsync(new RegionProfileDTO());
+            // Act
+            var result = await _regionController.GetRegionsBoardAsync();
+            var actual = (result as ObjectResult).Value;
+            // Assert
```
Comment (danylo-melnychuk) [inline EPlast/EPlast.Tests/Controllers/RegionsControllerTests.cs:497]: You shouldn't write your comment without code.
For example:
var actual = ...

//Assert
and your code.

File: EPlast/EPlast.Tests/Controllers/RegionsControllerTests.cs [hunk 4]
```diff
@@ -481,6 +482,24 @@ public async Task GetRegionsBoardAsync_ReturnsRegionsBoard()
...
+        [Test]
+        public async Task GetRegionsBoardAsync_GetDescription_string_empty()
+        {
+            // Arrange
+            _userManager
+                .Setup(x => x.GetUserAsync(new System.Security.Claims.ClaimsPrincipal())).ReturnsAsync(new User());
+            _regionService
+                .Setup(x => x.GetRegionByNameAsync(EnumExtensions.GetDescription(RegionsStatusType.RegionBoard), It.IsAny<User>()))
+                .ReturnsAsync(new RegionProfileDTO());
+            // Act
```
Comment (danylo-melnychuk) [inline EPlast/EPlast.Tests/Controllers/RegionsControllerTests.cs:494]: enter

File: EPlast/EPlast.Tests/Controllers/RegionsControllerTests.cs [hunk 5]
```diff
@@ -613,6 +632,203 @@ public async Task RemoveDocument_Id_ReturnsOkResult()
...
+        {
+            // Arrange
+            _userManager.Setup(x => x.GetUserAsync(It.IsAny<ClaimsPrincipal>())).ReturnsAsync(new User());
+            _userManager.Setup(x => x.GetRolesAsync(It.IsAny<User>())).ReturnsAsync(new List<string>() { "Admin" });
+            RegionAdministrationDTO admin = new RegionAdministrationDTO() { ID = 2 };
+
+            _logger.Setup(x => x.LogInformation(It.IsAny<string>()));
+
+            _regionAnnualReportService.Setup(x => x.DeleteAsync(It.IsAny<int>()));
+            // Act
```
Comment (danylo-melnychuk) [inline EPlast/EPlast.Tests/Controllers/RegionsControllerTests.cs:646]: Look comments below

File: EPlast/EPlast.Tests/Controllers/RegionsControllerTests.cs [hunk 6]
```diff
@@ -613,6 +632,203 @@ public async Task RemoveDocument_Id_ReturnsOkResult()
...
+            // Arrange
+            _userManager.Setup(x => x.GetUserAsync(It.IsAny<ClaimsPrincipal>())).ReturnsAsync(new User());
+            _userManager.Setup(x => x.GetRolesAsync(It.IsAny<User>())).ReturnsAsync(new List<string>() { "Admin" });
+
+            _regionAnnualReportService.Setup(x => x.EditAsync(It.IsAny<int>(), It.IsAny<RegionAnnualReportQuestions>()))
+                .ThrowsAsync(new InvalidOperationException());
+            
+
+            int reportID = 1;
+            // Act
```
Comment (danylo-melnychuk) [inline EPlast/EPlast.Tests/Controllers/RegionsControllerTests.cs:703]: Look comments below

File: EPlast/EPlast.Tests/Controllers/RegionsControllerTests.cs [hunk 7]
```diff
@@ -613,6 +632,203 @@ public async Task RemoveDocument_Id_ReturnsOkResult()
...
+        {
+            // Arrange
+            _userManager.Setup(x => x.GetUserAsync(It.IsAny<ClaimsPrincipal>())).ReturnsAsync(new User());
+            _userManager.Setup(x => x.GetRolesAsync(It.IsAny<User>())).ReturnsAsync(new List<string>() { "Admin" });
+            RegionAdministrationDTO admin = new RegionAdministrationDTO() { ID = 2 };
+
+            _logger.Setup(x => x.LogInformation(It.IsAny<string>()));
+      
+            int id = 0;
+            // Act
```
Comment (danylo-melnychuk) [inline EPlast/EPlast.Tests/Controllers/RegionsControllerTests.cs:759]: Put enter

File: EPlast/EPlast.Tests/Controllers/PasswordControllerTests.cs [hunk 8]
```diff
@@ -129,6 +129,41 @@ public async Task ChangePassword_Valid_Test()
             Assert.NotNull(result);
         }
 
+
+        [Test]
+        public async Task ChangePassword_BadRequest_refreshResult()
```
Comment (danylo-melnychuk) [inline EPlast/EPlast.Tests/Controllers/PasswordControllerTests.cs:134]: RefreshResult()

**Review by danylo-melnychuk — APPROVED** [review]

**Comment by danylo-melnychuk** [comment]
Good job

**Review by Oleh-Bodnar — APPROVED** [review]

**Review by xfreed — APPROVED** [review]

**Pull request merged**